codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2StreamChannel.java

```diff
@@ -68,13 +83,35 @@ public void run() {
         }
     };
 
-    // Volatile, as parent and child channel may be on different eventloops.
-    private volatile int streamId = -1;
+    private final Http2Stream2 stream;
     private boolean closed;
     private boolean readInProgress;
 
-    protected AbstractHttp2StreamChannel(Channel parent) {
+    /**
+     * The flow control window of the remote side i.e. the number of bytes this channel is allowed to send to the remote
+     * peer. The window can become negative if a channel handler ignores the channel's writability. We are using a long
+     * so that we realistically don't have to worry about underflow.
+     */
+    @SuppressWarnings("UnusedDeclaration")
+    private volatile long outboundFlowControlWindow;
+
+    static {
+        @SuppressWarnings("rawtypes")
+        AtomicLongFieldUpdater<AbstractHttp2StreamChannel> updater = AtomicLongFieldUpdater.newUpdater(
+                AbstractHttp2StreamChannel.class, "outboundFlowControlWindow");
+        if (updater == null) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Redundant nullcheck of updater, which is known to be non-null in io.netty.handler.codec.http2.AbstractHttp2StreamChannel.<static initializer for AbstractHttp2StreamChannel>() [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=findbugs%3ARCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE)